src/app/calendar/calendar.hpp

```diff
@@ -0,0 +1,18 @@
+#pragma once
+#include "../doctor/doctor.hpp"
+#include "../room/room.hpp"
+#include "connector.hpp"
+#include <chrono>
+#include <map>
+#include <memory>
+#include <stdexcept>
+
+class Calendar
+{
+    std::multimap<std::chrono::year_month_day, std::shared_ptr<Connector>> map_calendar_;
```

Comment:
With a container memory is already on heap, so you can keep it bit simpler by multimap of ymd and connectors

Reply:
Changed